### PR TITLE
[13_1_X]changing Era to Run3_2023 for 2023 WFs

### PR DESCRIPTION
**Pull request by sunilUIET**
#### PR description:

This PR is to change era from "Run3" to "Run3_2023" for relval WFs defined for 2023 conditions. Era "Run3_2023" is first introduced in PR https://github.com/cms-sw/cmssw/pull/41271 . 

#### PR validation:
Need to be tested with PR https://github.com/cms-sw/cmssw/pull/41271

### CODE REVIEW

**Comment by sunilUIET** [comment]
please test with #41271

**Comment by francescobrivio** [comment]
Hi @sunilUIET thanks for this!
Just for my understanding: is there a plan to update also all the `'--era':'Run3'` occurrences in [Configuration/PyReleaseValidation/python/relval_steps.py](https://github.com/cms-sw/cmssw/blob/master/Configuration/PyReleaseValidation/python/relval_steps.py) (and all the other release validation scripts)?

**Comment by sunilUIET** [comment]
> Hi @sunilUIET thanks for this! Just for my understanding: is there a plan to update also all the `'--era':'Run3'` occurrences in [Configuration/PyReleaseValidation/python/relval_steps.py](https://github.com/cms-sw/cmssw/blob/master/Configuration/PyReleaseValidation/python/relval_steps.py) (and all the other release validation scripts)?



> Configuration/PyReleaseValidation/python/relval_steps.py

Hi @francescobrivio , I think we can change it for 2023 WFs only, for the time being, keeping the settings as it is for old WFs.  As far as 2023 WFs are concerned, IIUC, it should be enough to change  settings in https://github.com/cms-sw/cmssw/blob/master/Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py#L2364,L2370 and changes should be propagated to all steps.

**Comment by sunilUIET** [comment]
test parameters:

- relvals_opt= -w upgrade
- workflows = 12402.0

**Comment by sunilUIET** [comment]
@cmsbuild please test with https://github.com/cms-sw/cmssw/pull/41271

**Comment by srimanob** [comment]
Not sure I understand this PR. First comment in https://github.com/cms-sw/cmssw/pull/41271#issuecomment-1505500629. But what I don't understand in this PR is that why do you make new non-default workflow (default = .0). Whatever done with Run3_2023 should be default, or I miss some context? thx.

**Comment by sunilUIET** [comment]
Hi @srimanob , It is unclear if this has to be the default. Technically, we had a problem in updating  2023 WFs with era Run3_2023 (first commit of this PR). So, we took a call to add new WFs with this era setting so that #41271 can have WFs to test with. I agree with your comment in #41271 which will keep the latest settings by default.

**Comment by srimanob** [comment]
It we want to do some alternative to test, I propose to make Era to be more specific, i.e. the same as we did for mkfit. Then we use that with workflow+prefix. Run3_2023 is not a good naming if we don't want it to be default. FYI @swagata87

**Comment by srimanob** [comment]
Hi @sunilUIET 

You will need to update this line, https://github.com/cms-sw/cmssw/blob/master/Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py#L2239
by adding 
`and '2023' not in stepDict[step][k]['--era']`
This is to protect the wierd era modification. Then everything should work out-of box without the need for extra workflow.

**Comment by sunilUIET** [comment]
@cmsbuild please test with https://github.com/cms-sw/cmssw/pull/41271

**Comment by sunilUIET** [comment]
+pdmv

**Comment by srimanob** [comment]
+Upgrade

File: Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py [hunk 1]
```diff
@@ -2185,7 +2186,7 @@ def condition(self, fragment, stepList, key, hasHarvest):
 
 class UpgradeWorkflow_DDDDB(UpgradeWorkflow):
     def setup_(self, step, stepName, stepDict, k, properties):
-        if 'Run3' in stepDict[step][k]['--era'] and 'Fast' not in stepDict[step][k]['--era']:
+        if 'Run3' in stepDict[step][k]['--era']  and '2023' not in stepDict[step][k]['--era'] and 'Fast' not in stepDict[step][k]['--era']:
```
Comment (perrotta) [inline Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py:2189]: (Just to understand) What is the purpose of this modification?
Should we add 2024, 2025, etc. here in the coming years?

Comment (srimanob) [reply Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py:2189]: This is just a temporary fix to not screw up on modification. The follow up PR should clean this thing out when we have a clear Era, and all necessary Era(s) are in place, i.e. DDD for 2023,4,5 or FastSim for 2023,4,5.

**Comment by swagata87** [comment]
do we need a backport of this PR to 13_0_X? 

FYI, https://github.com/cms-sw/cmssw/pull/41368 this is backport of my PR

**Pull request merged**